WordPress/Classes/ViewRelated/Blog/NoResultsViewController.swift

```diff
@@ -55,7 +55,13 @@ import Reachability
     }
 
     private var buttonText: String?
-    private var imageName: String?
+    private var imageName: String? {
+        didSet {
+            if isConnectionAvailable == false {
```

Comment:
This pattern seems to be repeated multiple times in the code now. My concern is it might be harder to scan the code if we ask this question in the future: _"What happens to the view if the device is offline?"_. 

From what I can tell, there are only 2 entry points for modifying the properties:

- `@objc class func controllerWith(title: String,...`
- `@objc func configure(title: String,...`

What do you think about adding this logic in there instead? Example:

```swift
        let isReachable = reachability?.isReachable()

        titleText = isReachable ? title : NoConnectionMessages.title
        subtitleText = isReachable ? subtitle : NoConnectionMessages.subTitle
        attributedSubtitleText = isReachable ? attributedSubtitle : NoConnectionMessages.subTitle
        buttonText = buttonTitle
        imageName = isReachable ? image : "cloud"
```

There are still repetitions but they're colocated and easier to find. 

The contents of `@objc class func controllerWith()` can be changed to use `configure()` to eliminate redundancy:

```swift
    @objc class func controllerWith(title: String,
                                    buttonTitle: String? = nil,
                                    subtitle: String? = nil,
                                    attributedSubtitle: NSAttributedString? = nil,
                                    image: String? = nil,
                                    subtitleImage: String? = nil,
                                    accessoryView: UIView? = nil) -> NoResultsViewController {
        let controller = NoResultsViewController.controller()
        controller.configure(title: title,
                buttonTitle: buttonTitle,
                subtitle: subtitle,
                attributedSubtitle: attributedSubtitle,
                image: image,
                subtitleImage: subtitleImage,
                accessoryView: accessoryView)
        return controller
    }
```

With this change, `configure()` will be the source of truth of what happens if the device is offline.

Reply:
I like this suggestions a lot and makes sense. Im going to update the base branch and the image part for this branch. Thank you so much for this